bindings/python/src/connection.rs

```diff
@@ -1,50 +1,42 @@
 use std::sync::Arc;
 
-use datafusion::logical_expr::LogicalPlan as DFLogicalPlan;
-use datafusion_ext::vars::SessionVars;
-use futures::lock::Mutex;
 use once_cell::sync::OnceCell;
 use pyo3::prelude::*;
 use pyo3::types::PyType;
-use sqlexec::engine::{Engine, EngineStorage, SessionStorageConfig, TrackedSession};
-use sqlexec::{LogicalPlan, OperationInfo};
-
-use crate::execution_result::PyExecutionResult;
-
-pub(super) type PyTrackedSession = Arc<Mutex<TrackedSession>>;
```

Comment:
In a lot of ways, this `Arc<Mutex<>>` thing here serves the same purpose that the `Arc<Mutex<>>` around the `Operation` in the `PyExecution` object.